skills/csharp/tests/todoskill.tests/Flow/ToDoSkillTestBase.cs

```diff
@@ -183,5 +185,28 @@ protected string[] ActionEndMessage()
         {
             return GetTemplates(ToDoMainResponses.CompletedMessage);
         }
+
+        protected Action<IActivity> CheckForEoC(int taskCount)
+        {
+            return activity =>
+            {
+                var eoc = (Activity)activity;
+                Assert.AreEqual(ActivityTypes.EndOfConversation, eoc.Type);
+
+                if (eoc.Value is ActionResult)
```

Comment:
If for testing, should assert the type of eoc.Value? Or if return a wrong type, we don't know

Reply:
User should indicate the type of eoc.Value when calling CheckForEoC like CheckForEoC(Enum valueType, int taskCount)

For current code, if one flow returns ActionResult while actually we expect TodoListInfo, it is not checked.

Reply:
Sure. 
Is Enum needed here? Or we just use CheckForEoC(Type valueType, int taskCount)? Because we only have two types...

Reply:
No, it is just an example. Any kind of indicator is fine